x/config/chooser.go

```diff
@@ -77,17 +77,21 @@ func (pc PeerList) BuildPeerList(transport peer.Transport, identify func(string)
 			continue
 		}
 
-		var peerListUpdaterConfig attributeMap
-		if _, err := c.Etc.Pop(peerListName, &peerListUpdaterConfig); err != nil {
+		var peerListConfig attributeMap
```

Comment:
This code can be greatly simplified if the peer list configuration is no longer shared with the outbound options.

The only thing remaining in `c.Etc` at this point will be a single entry: the peer list name and its entries. So you can just assert that `len(c.Etc) == 1` and then,

```
var peerListName string
for peerListName = range c.Etc {
  break
}

c.Etc.Pop(peerListName, ...)
```

The `c.Etc` length afterwards won't be necessary because we know there's only one entry -- which we removed.